### PR TITLE
Migrate basic CI/CD pipeline to GitHub Actions

### PR DESCRIPTION
**Pull request by innomaxx**

### CODE REVIEW

**Comment by andrii-bodnar** [comment]
@innomaxx is this PR finished?

**Comment by innomaxx** [comment]
> @innomaxx is this PR finished?

No, still working. I add steps one by one to ensure it works properly. I will notify you on completion

**Comment by andrii-bodnar** [comment]
The build job should include tests and there should be an additional job (not matrix) to lint, run tests once, and generate a code coverage report

We just do not need this report for all matrix runs

**Comment by innomaxx** [comment]
@andrii-bodnar done

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    types: [created]
+    
+env:
+  PROJECT_PATH: src/Crowdin.Api
+  PROJECT_CSPROJ_PATH: ${{ env.PROJECT_PATH }}/Crowdin.Api.csproj
+  
+jobs:
+  publish-nuget:
+    runs-on: ubuntu-latest
+    permissions:
```
Comment (andrii-bodnar) [inline .github/workflows/publish.yml:14]: we do not need these permissions here since we do not publish any GitHub packages

Comment (innomaxx) [reply .github/workflows/publish.yml]: fixed

File: .github/workflows/basic.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
+name: Build, tests, code coverage
```
Comment (andrii-bodnar) [inline .github/workflows/basic.yml:1]:
```suggestion
name: Tests
```

File: .github/workflows/basic.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        
+      - name: Build
+        run: dotnet build -c Release --no-restore
+        
+      - name: Test
+        run: dotnet test --no-restore --verbosity normal
+
+  code-coverage:
+    runs-on: ubuntu-latest
+
```
Comment (andrii-bodnar) [inline .github/workflows/basic.yml:43]:
```suggestion
    needs: build
```

**Review by andrii-bodnar — APPROVED** [review]

**Pull request merged**